Description: add static library build
 Add upstream missed static library target.
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Last-Update: 2021-12-04

---

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -249,6 +249,25 @@ if(BUILD_SHARED_LIBS)
   set_target_properties(snappy PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 endif(BUILD_SHARED_LIBS)
 
+ADD_LIBRARY(snappy-static STATIC
+        snappy-c.cc
+        snappy-c.h
+        snappy-sinksource.cc
+        snappy-sinksource.h
+        snappy-stubs-internal.cc
+        snappy-stubs-public.h
+        snappy.cc
+        snappy.h)
+
+target_include_directories(snappy-static
+  PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
+
+SET_TARGET_PROPERTIES(snappy-static PROPERTIES OUTPUT_NAME snappy)
+
 if(SNAPPY_BUILD_TESTS OR SNAPPY_BUILD_BENCHMARKS)
   add_library(snappy_test_support "")
   target_sources(snappy_test_support
@@ -359,7 +378,7 @@ endif(SNAPPY_FUZZING_BUILD)
 include(GNUInstallDirs)
 
 if(SNAPPY_INSTALL)
-  install(TARGETS snappy
+  install(TARGETS snappy snappy-static
     EXPORT SnappyTargets
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
